src/components/project_admin/ag_suites/ag_suites.vue

```diff
@@ -0,0 +1,608 @@
+        }
+      }
+    }
+    return false;
+  }
+
+  get next_command_is_available() {
+    let next_suite: AGTestSuite;
+    let next_case: AGTestCase;
+
+    if (this.index_active_command !== -1
+        && this.index_active_case !== -1
+        && this.index_active_suite !== -1) {
+
+      let num_cases = this.active_suite!.ag_test_cases.length;
+
+      // if there is a command at the same index in the next case
+      if (num_cases - 1 > this.index_active_case) {
+        next_case = this.active_suite!.ag_test_cases[this.index_active_case + 1];
+        return next_case.ag_test_commands.length > this.index_active_command;
+      }
+      // if you can go to the next suite
+      else if (this.index_active_suite < this.test_suites.length - 1) {
+        next_suite = this.test_suites[this.index_active_suite + 1];
+        if (next_suite.ag_test_cases.length > 0) {
+          next_case = next_suite.ag_test_cases[0];
+          return next_case.ag_test_commands.length > this.index_active_command;
+        }
+      }
+    }
+    return false;
+  }
+
+  go_to_next_command() {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase;
+
+    let command_index = this.index_active_command;
+    if (this.active_suite!.ag_test_cases.length - 1 > this.index_active_case) {
+      parent_case = parent_suite!.ag_test_cases[this.index_active_case + 1];
+    }
+    else {
+      parent_suite = this.test_suites[this.index_active_suite + 1];
+      parent_case = parent_suite!.ag_test_cases[0];
+    }
+    this.update_active_command(parent_case!.ag_test_commands[command_index],
+                               parent_case,
+                               parent_suite);
+  }
+
+  go_to_prev_command() {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase;
+
+    let command_index = this.index_active_command;
+    if (this.index_active_case !== 0) {
+      parent_case = parent_suite.ag_test_cases[this.index_active_case - 1];
+    }
+    else {
+      parent_suite =  this.test_suites[this.index_active_suite - 1];
+      let num_cases = parent_suite.ag_test_cases.length;
+      parent_case = parent_suite.ag_test_cases[num_cases - 1];
+    }
+    this.update_active_command(parent_case.ag_test_commands[command_index],
+                               parent_case,
+                               parent_suite);
+  }
+
+  get active_level_is_suite() {
+    return this.active_suite !== null && this.active_case === null;
+  }
+
+  get active_level_is_command() {
+    return this.active_command !== null;
+  }
+
+  update_active_suite(ag_suite: AGTestSuite | null) {
+    this.reset_active_indices();
+    if (ag_suite !== null) {
+      this.index_active_suite = this.test_suites.findIndex(
+        (test_suite: AGTestSuite) => test_suite.pk === ag_suite.pk);
+    }
+  }
+
+  update_active_case(ag_case: AGTestCase, parent_suite: AGTestSuite) {
```

Comment:
What if we had these update_active_xx functions only take in one parameter and then search for the parent(s)? Note that an ag test case knows the pk of its parent (similarly with commands)